### PR TITLE
SSCS-2223 DWP issuing office

### PR DESCRIPTION
**Pull request by bendiggle**
- Add DWP Issuing office page
- Add file with list of office Id's that are allowed
- Edit the benefit Type class so that the next step goes to the new page when PIP is the benefit type. Otherwise go to mrn-date
- Add unit tests and e2e tests for the new page
- Edit the benefit type unit tests for the next step

### CODE REVIEW

File: steps/compliance/dwp-issuing-office/DWPIssuingOffice.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const { Question, goTo } = require('@hmcts/one-per-page');
+const { form, textField } = require('@hmcts/one-per-page/forms');
+const { numbers } = require('utils/regex');
+const officeIds = require('./dwpIssuingOfficeIds');
```
Comment (paulgain) [inline steps/compliance/dwp-issuing-office/DWPIssuingOffice.js]: For consistency can we add full paths please.

File: steps/compliance/dwp-issuing-office/DWPIssuingOffice.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class DWPIssuingOffice extends Question {
+
+    static get path() {
+
+        return paths.compliance.dwpIssuingOffice;
+    }
+
+    get form() {
+
+        return form(
```
Comment (paulgain) [inline steps/compliance/dwp-issuing-office/DWPIssuingOffice.js]: Space

File: steps/compliance/dwp-issuing-office/content.en.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      "header": "Enter the number from the top of DWP’s address",
+      "label": "Personal Independence Payment",
+      "error": {
+        "required": "Enter the DWP issuing office",
+        "notNumeric": "DWP issuing office must be a number",
+        "invalid": "Enter a valid DWP issuing office"
+      }
+    }
+  },
+  "noMRN": "I don’t have a Mandatory Reconsideration Notice"
```
Comment (paulgain) [inline steps/compliance/dwp-issuing-office/content.en.json]: Can you move this above the fields please.

File: test/unit/steps/compliance/dwp-issuing-office/DWPIssuingOffice.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    describe('get form()', () => {
+
+        let fields;
+        let field;
+
+        beforeEach(() => {
+            fields = dWPIssuingOffice.form.fields;
+        });
+
+        after(() => {
```
Comment (paulgain) [inline test/unit/steps/compliance/dwp-issuing-office/DWPIssuingOffice.test.js]: There's no need to define `after` in each of our tests, moreover, I've removed all of these from our tests. I'm unsure if the GC will run during a test cycle, even if it did the amount of memory freed would be negligible considering the number of tests we have.

Comment (bendiggle) [reply test/unit/steps/compliance/dwp-issuing-office/DWPIssuingOffice.test.js]: @paulgain after still exists in all the tests for the fields. If you want, I can remove them all

Comment (paulgain) [reply test/unit/steps/compliance/dwp-issuing-office/DWPIssuingOffice.test.js]: Yes please, I have removed quite a few. Feel free to clear up. Thank you.

File: steps/compliance/dwp-issuing-office/DWPIssuingOffice.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+            textField('pipNumber')
+                .joi(
```
Comment (paulgain) [inline steps/compliance/dwp-issuing-office/DWPIssuingOffice.js]: Sorry, only just seen this. Can you move the `.joi` up a line so we are consistent with all other steps. AppellantContactDetails is a good example.

**Pull request merged**